frontend/src/pages/fb/[fid].tsx

```diff
@@ -42,23 +42,36 @@ const Page = () => {
   const { fid } = router.query;
   const { loading, error, data } = useQuery(GET_FISHBOWL, { variables: { slug: fid } });
 
+  const handleJoinAsGuest = () => {
+    setJoinAsGuest(true);
+  };
+
+  const shoulPrintPreJoinPage: boolean = (joinAsGuest || isAuthenticated) && prejoin;
+  const shoulPrintFishbowlPage: boolean = fishbowlReady && (isAuthenticated || isGuest);
+
+  useEffect(() => {
+    router.beforePopState(({ as }) => {
```

Comment:
`as` is a string?
https://nextjs.org/docs/api-reference/next/router

Reply:
Yep and it comes typed from next.